TelemetrySite/client/src/contextForm/ContextForm.jsx

```diff
@@ -0,0 +1,452 @@
+  });
+  //current selected value of each dropdown
+  const [configSelects, setConfigSelects] = useState({
+    bms: null,
+    imu: null,
+    tmu: null,
+    tms: null,
+    pvc: null,
+    mc: null,
+  });
+  //values passed from back end
+  const [dropDownOptions, setDropdownOptions] = useState({
+    bms: [],
+    imu: [],
+    tmu: [],
+    tms: [],
+    pvc: [],
+    mc: [],
+    bike: [],
+  });
+
+  const [bikeSelected, setBikeSelected] = useState(false);
+
+  /* -------------------------------------------------------------------------- */
+  /* --------------------------- Initialize Constants ------------------------- */
+  /* -------------------------------------------------------------------------- */
+  let ConfigName = ["bms", "imu", "tmu", "tms", "pvc", "mc"];
+
+  //which config selects are optional
+  let RequiredSelects = {
+    bmsConfig: true,
+    tmsConfig: true,
+    imuConfig: false,
+    tmuConfig: false,
+    pvcConfig: true,
+    mcConfig: true,
+  };
+  let FormId = "ContextForm";
+
+  let navigate = useNavigate();
+  /* -------------------------------------------------------------------------- */
+  /* ----------------------------- Const Functions ---------------------------- */
+  /* -------------------------------------------------------------------------- */
+
+  /**
+   * When a select field for the config forms updates,
+   * pass the new value here to update the useState hook
+   * and rerender effected components
+   *
+   * @param {string} configName - Name of config to update
+   * @param {string} value - New value of select field
+   */
+  const handleConfigSelectChange = (configName, value) => {
+    setConfigSelects((prev) => ({ ...prev, [configName]: value }));
+  };
+
+  /**
+   * Whenever the select field on a config form changes
+   * check if it is Custom. If it is, display form elements.
+   * Else, set form object to null
+   *
+   * @param {formElement.event} event - event that occurred to the select element
+   * @param {string} configName - name of config being updated
+   */
+  const handleConfigFormChange = (event, configName) => {
+    const value = event.target.value;
+    //handle the bike differently from the configs
+    if (configName !== "bike") {
+      handleConfigSelectChange(configName, value);
+      if (value === "Custom") {
+        const formElement = GenerateFormElement(`${configName}Config`);
+        setFormElements((prev) => ({ ...prev, [configName]: formElement }));
+      } else {
+        setFormElements((prev) => ({ ...prev, [configName]: null }));
+      }
+    } else {
+      if (value === "Custom") {
+        UpdateBikeForm(GenerateFormElement("BikeConfig"));
+        setBikeSelected(true);
+      } else {
+        UpdateBikeForm(null);
+        setBikeSelected(false);
+      }
+    }
+  };
+
+  /**
+   * Call to the python API. For each of the config
+   * forms, get any saved configs that are in the DB.
+   * Set them to the corresponding dropdown option
```

Comment:
Not a huge deal, but your line limit for comments is inconsistent. We should look at getting an auto-formatter after the prototype is good to go